shimmingtoolbox/unwrap/unwrap_phase.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+""" Wrapper to different unwrapping algorithms
+
+"""
+
+
+import numpy as np
+from shimmingtoolbox.unwrap.prelude import prelude
+
+
+def unwrap_phase(complex_array, affine, unwrap_function="prelude", mask=np.array([-1])):
+    """ Calls different unwrapping algorithms according to the specified `unwrap_function` parameter
+
+    Args:
+        complex_array (numpy.ndarray): 2D,3D,4D or 5D complex values numpy array to perform phase unwrapping
+        affine (numpy.ndarray): 2D array (4x4) containing the transformation coefficients. Can be acquired by :
+            nii = nib.load("nii_path")
+            affine = nii.affine
+        unwrap_function (string, optional): unwrapper algorithm name. Possible values : prelude
+        mask: numpy array of booleans with shape of `complex_array` to mask during phase unwrapping. Calling `mask` wth
+        np.array([-1]) is the same as specifying no mask.
+
+    Returns:
+        numpy.ndarray: 3D array with the shape of `complex_array` of the unwrapped phase output from prelude
+
+    Raises:
+        Exception: complex array does not have 2,3,4,5 dimensions
+        Exception: unwrapping algorithm in `unwrap_function` is not supported
+    """
+    unwrapped_phase = np.empty(complex_array.shape, np.complex)
+    if unwrap_function == "prelude":
```

Comment:
If this part is meant to have multiple unwrap_function, a better practice would be to create separate functions for each unwrap_function and save them in a dictionary with the unwrap_function as a key. Then in the function unwrap_phase you would just need to call it with the correct arguments: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/18#discussion_r453138872
```suggestion
unwrap_functions = {
    'prelude' = unwrap_prelude,
    'other unwrap' = unwrap_other,
}

unwrapped_phase = unwrap_functions[unwrap_function](complex_array, affine, mask)
```
Also the dimension checks should be in the prelude function instead of here, the user function should not be aware of the different ways to give arguments. It should only give the inputs necessary and if the format is not adequate, then return an exception. Also, if someone wants only to use the prelude function for XY reason, he would have to redo all these checks and it results in duplicated code.

Reply:
If we add another unwrapper then, we would have to rewritte these checks for the other unwrapper function. So another solution is to reshape the complexe_array beforehand and pass it to the function and adapt the function so it can take any of these shapes. I'll think of how later on.

Reply:
From what I remember, previous discussions settled on having unwrappers that have as inputs (x,y,z) and dealing with echoes and acqs in the parent function. II'm open to revisit the question though.
[Comment](https://github.com/shimming-toolbox/shimming-toolbox/pull/131#discussion_r443025788) I'm referring to

Reply:
I see. Then, we could still do the dimension check here, but still, my first comment on the function caller could be easier to avoid having other ifs for the different call to unwrappers ('prelude', etc.). [https://github.com/shimming-toolbox/shimming-toolbox-py/pull/15/files/d50e95663d36e378ffd12b30c2532a6e87ee303b#r454710944](url)

Reply:
unwrap_phase is deprecated, prelude will be run directly from the script so we don't need to bother.